src/Dao/EnwikiDao.php

```diff
@@ -75,6 +97,37 @@ public function getUserDetails( $diff ) {
 
 
 	/**
+	 * Determine which of the given pages are dead
+	 *
+	 * @param $titles array Page titles
+	 * @return array the pages that are dead
+	 */
+	public function getDeadPages( $titles ) {
+		if ( !$titles ) {
+			return false;
```

Comment:
Should probably either return an empty array or you should change the phpdoc for the return value to `@return array|bool`